tool/tsh/db.go

```diff
@@ -850,6 +851,10 @@ func getDatabaseInfo(cf *CLIConf, tc *client.TeleportClient, dbName string) (*tl
 	if err != nil && !trace.IsNotFound(err) {
 		return nil, nil, trace.Wrap(err)
 	}
+	if dbName == "" {
+		return nil, nil, trace.NotFound(
+			"No database name given or detected. Provide a database name as an argument\n       or use 'tsh db login [<flags>] <db>' prior to attempting to connect")
```

Comment:
Nice UX improvement! Since these `tsh db` commands also have `--db-name`, it may help to avoid confusion by changing the error message to:
```suggestion
			"No database given or detected. Provide a database as an argument\n       or use 'tsh db login [<flags>] <db>' prior to attempting to connect")
```